### ISSUE TITLE
eager_instantiation branch instatiation issues

### ISSUE THREAD

**Issue by afarooqui98** [afarooqui98/gem5_GUI#86]
[nocachem5out.zip](https://github.com/afarooqui98/ECS193a/files/4361928/nocachem5out.zip)

attached is the m5out directory for the nocache example. The flow of the instantiation stars from the catalog view. When the double click handler is triggered, a new sym_object is created, which is where the SimObject instantiation function is invoked. This then goes to m5_calls, specifically `instantiate_object`, which invokes the constructor and then runs enumerateParams on the instance of the SimObject. Flow is therefore catalog_view.py (createSymObject) -> m5_calls (instantiateSimObject, traverse_hierarchy.

**Comment by powerjg** [comment]
What branch should I use to try to replicate this issue?

**Comment by powerjg** [comment]
I don't see anything obvious in the config.ini/json that's wrong... I'll need to do some debugging with the actual code :)

**Comment by powerjg** [comment]
Can you point me to what branch to use to reproduce this issue? Is it the master branch?

**Comment by afarooqui98** [comment]
Sorry about that! eager_instantiation is the branch that should be used to replicate the issue. I didn't want to merge anything to master so I kept the changes in that branch

**Comment by powerjg** [comment]
Thanks! I'll try to take a look at this ASAP.

**Comment by powerjg** [comment]
OK. Sorry for taking so long here. There's been a lot of high priority things on my plate. I can't remember how to recreate this issue. Could you either describe how to reproduce in more detail, or we could set up a quick zoom today or tomorrow before our meeting.

**Comment by Ravishdeep10** [comment]
Basically it is just creating the simple_system_no_cache example in the GUI and when we instantiate the bug occurs. Ahmed and I can do a quick meeting before the main one tomorrow as well.

**Comment by powerjg** [comment]
Let's just talk about it in the meeting today. I just tried to reproduce the problem, but ran into a number of other issues.

I have a feeling the problem is that we need to flesh out some other "features" before the x86 example will work. Maybe trying RISC-V or a traffic generator example would be better.

**Comment by powerjg** [comment]
FYI, the problem is 

```
Getting value for X86TLB eventq_index                                                                                                                 Parent.eventq_index 
```

I'm going to keep digging, but I wanted you to know I've started getting closer :)

**Comment by powerjg** [comment]
AHA! I figured this out!!!!!

The problem was that the CPU has a ton of these "implicit" objects, and they aren't created correctly. I'm not sure what the "long term" solution is, but in the short term, if you add an `itb` (type TLB), `dtb` (type TLB), `tracer` (type ExeTracer), `isa` (type ISA), and `workload` things "work".

I've attached a ui file that "works" or at least gets past this problem for RISC-V. I switched to RISC-V to try to skip some of the x86 BS. (Rename the file to *.ui, github won't let me upload unless it ends in .txt)

[testing.txt](https://github.com/afarooqui98/ECS193a/files/4463714/testing.txt)
